internal/repositories/devicedefinition/devicedefinition.go

```diff
@@ -0,0 +1,224 @@
+package devicedefinition
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	gmodel "github.com/DIMO-Network/identity-api/graph/model"
+	"github.com/DIMO-Network/identity-api/internal/helpers"
+	"github.com/DIMO-Network/identity-api/internal/repositories/base"
+	"github.com/vektah/gqlparser/v2/gqlerror"
+	"golang.org/x/exp/slices"
+	"net/http"
+	"net/url"
+	"path"
+	"strings"
+)
+
+// TokenPrefix is the prefix for a global token id for Device Definition.
+const TokenPrefix = "DD"
+
+type DeviceDefinitionRepository struct {
+	*base.Repository
+}
+
+func ToAPI(v *DeviceDefinitionTablelandModel) *gmodel.DeviceDefinition {
+	var result = gmodel.DeviceDefinition{}
+
+	result.ID = v.ID
+	result.Ksuid = &v.KSUID
+	result.Year = &v.Year
+	result.Model = &v.Model
+
+	if v.Metadata.DeviceAttributes != nil && len(v.Metadata.DeviceAttributes) > 0 {
```

Comment:
You do not need this check; in Golang, it is safe to range over a nil or empty list.
```suggestion
```